### ISSUE TITLE
Events on contracts

### ISSUE THREAD

**Issue by jennazenk** [melonproject/protocol#543]
- [x]  Created event after successful share creation (as Transfer(0x0, investor) from ERC20
- [x] RequestExecuted (after requestInvestment)
- [x] OrderUpdated (after callOnExchange)

**Comment by travs** [comment]
- [ ]  Created event after successful share creation
- [ ] RequestExecuted (after requestInvestment)
- [ ] OrderUpdated (after callOnExchange)

Old events we may want to consider
```
legacy/contracts/dependencies/weth.sol:6:    event Join(address indexed dst, uint wad);
legacy/contracts/dependencies/weth.sol:7:    event Exit(address indexed src, uint wad);
legacy/contracts/dependencies/medianizer.sol:6:    event LogValue(bytes32 val);
legacy/contracts/dependencies/test.sol:17:    event eventListener          (address target, bool exact);
legacy/contracts/dependencies/test.sol:18:    event logs                   (bytes);
legacy/contracts/dependencies/test.sol:19:    event log_bytes32            (bytes32);
legacy/contracts/dependencies/test.sol:20:    event log_named_address      (bytes32 key, address val);
legacy/contracts/dependencies/test.sol:21:    event log_named_bytes32      (bytes32 key, bytes32 val);
legacy/contracts/dependencies/test.sol:22:    event log_named_decimal_int  (bytes32 key, int val, uint decimals);
legacy/contracts/dependencies/test.sol:23:    event log_named_decimal_uint (bytes32 key, uint val, uint decimals);
legacy/contracts/dependencies/test.sol:24:    event log_named_int          (bytes32 key, int val);
legacy/contracts/dependencies/test.sol:25:    event log_named_uint         (bytes32 key, uint val);
legacy/contracts/dependencies/test.sol:44:        eventListener(target, true);
legacy/contracts/dependencies/group.sol:29:    event Proposed   (uint id, bytes calldata);
legacy/contracts/dependencies/group.sol:30:    event Confirmed  (uint id, address member);
legacy/contracts/dependencies/group.sol:31:    event Triggered  (uint id);
legacy/contracts/dependencies/auth.sol:23:    event LogSetAuthority (address indexed authority);
legacy/contracts/dependencies/auth.sol:24:    event LogSetOwner     (address indexed owner);
legacy/contracts/dependencies/note.sol:19:    event LogNote(
legacy/contracts/dependencies/erc20.sol:12:    event Approval(address indexed src, address indexed guy, uint wad);
legacy/contracts/dependencies/erc20.sol:13:    event Transfer(address indexed src, address indexed dst, uint wad);
legacy/contracts/competitions/CompetitionInterface.sol:9:
legacy/contracts/competitions/CompetitionInterface.sol:10:    event Register(uint withId, address fund, address manager);
legacy/contracts/competitions/CompetitionInterface.sol:11:    event ClaimReward(address registrant, address fund, uint shares);
legacy/contracts/competitions/TestCompetition.sol:65:
legacy/contracts/competitions/TestCompetition.sol:66:    event Register(uint withId, address fund, address manager);
legacy/contracts/competitions/Competition.sol:64:
legacy/contracts/competitions/Competition.sol:65:    event Register(uint withId, address fund, address manager);
legacy/contracts/assets/ERC223Interface.sol:10:    event Transfer(address indexed from, address indexed to, uint value, bytes data);
legacy/contracts/assets/SharesInterface.sol:8:
legacy/contracts/assets/SharesInterface.sol:9:    event Created(address indexed ofParticipant, uint atTimestamp, uint shareQuantity);
legacy/contracts/assets/SharesInterface.sol:10:    event Annihilated(address indexed ofParticipant, uint atTimestamp, uint shareQuantity);
legacy/contracts/assets/AssetInterface.sol:18:    event Approval(address indexed _owner, address indexed _spender, uint _value);
legacy/contracts/system/StakingInterface.sol:4:
legacy/contracts/system/StakingInterface.sol:5:    event Staked(address indexed user, uint256 amount, uint256 total, bytes data);
legacy/contracts/system/StakingInterface.sol:6:    event Unstaked(address indexed user, uint256 amount, uint256 total, bytes data);
legacy/contracts/system/StakingInterface.sol:7:    event StakeBurned(address indexed user, uint256 amount, bytes data);
legacy/contracts/assets/ERC20Interface.sol:8:
legacy/contracts/assets/ERC20Interface.sol:9:    event Transfer(address indexed from, address indexed to, uint tokens);
legacy/contracts/assets/ERC20Interface.sol:10:    event Approval(address indexed tokenOwner, address indexed spender, uint tokens);
legacy/contracts/system/OperatorStaking.sol:15:
legacy/contracts/system/OperatorStaking.sol:16:    event Staked(address indexed user, uint256 amount, uint256 total, bytes data);
legacy/contracts/system/OperatorStaking.sol:17:    event Unstaked(address indexed user, uint256 amount, uint256 total, bytes data);
legacy/contracts/system/OperatorStaking.sol:18:    event StakeBurned(address indexed user, uint256 amount, bytes data);
legacy/contracts/system/Governance.sol:32:
legacy/contracts/system/Governance.sol:33:    event VersionUpdated(uint id);
legacy/contracts/exchange/adapter/SimpleAdapter.sol:19:
legacy/contracts/exchange/adapter/SimpleAdapter.sol:20:    event OrderUpdated(uint id);
legacy/contracts/exchange/adapter/SimpleAdapterWithApprove.sol:19:
legacy/contracts/exchange/adapter/SimpleAdapterWithApprove.sol:20:    event OrderUpdated(uint id);
legacy/contracts/exchange/GenericExchangeInterface.sol:13:
legacy/contracts/exchange/GenericExchangeInterface.sol:14:    event OrderUpdated(uint id);
legacy/contracts/exchange/thirdparty/EtherDelta.sol:57:
legacy/contracts/exchange/thirdparty/EtherDelta.sol:58:  event Transfer(address indexed _from, address indexed _to, uint256 _value);
legacy/contracts/exchange/thirdparty/EtherDelta.sol:59:  event Approval(address indexed _owner, address indexed _spender, uint256 _value);
legacy/contracts/exchange/thirdparty/EtherDelta.sol:161:
legacy/contracts/exchange/thirdparty/EtherDelta.sol:162:  event Order(address tokenGet, uint amountGet, address tokenGive, uint amountGive, uint expires, uint nonce, address user);
legacy/contracts/exchange/thirdparty/EtherDelta.sol:163:  event Cancel(address tokenGet, uint amountGet, address tokenGive, uint amountGive, uint expires, uint nonce, address user, uint8 v, bytes32 r, bytes32 s);
legacy/contracts/exchange/thirdparty/EtherDelta.sol:164:  event Trade(address tokenGet, uint amountGet, address tokenGive, uint amountGive, address get, address give);
legacy/contracts/exchange/thirdparty/EtherDelta.sol:165:  event Deposit(address token, address user, uint amount, uint balance);
legacy/contracts/exchange/thirdparty/EtherDelta.sol:166:  event Withdraw(address token, address user, uint amount, uint balance);
legacy/contracts/exchange/thirdparty/0x/TokenTransferProxy.sol:53:
legacy/contracts/exchange/thirdparty/0x/TokenTransferProxy.sol:54:    event Transfer(address indexed _from, address indexed _to, uint _value);
legacy/contracts/exchange/thirdparty/0x/TokenTransferProxy.sol:55:    event Approval(address indexed _owner, address indexed _spender, uint _value);
legacy/contracts/exchange/thirdparty/0x/TokenTransferProxy.sol:106:
legacy/contracts/exchange/thirdparty/0x/TokenTransferProxy.sol:107:    event LogAuthorizedAddressAdded(address indexed target, address indexed caller);
legacy/contracts/exchange/thirdparty/0x/TokenTransferProxy.sol:108:    event LogAuthorizedAddressRemoved(address indexed target, address indexed caller);
legacy/contracts/exchange/thirdparty/PersonalMarket.sol:19:    event LogItemUpdate(uint id);
legacy/contracts/exchange/thirdparty/PersonalMarket.sol:20:    event LogTrade(uint pay_amt, address indexed pay_gem,
legacy/contracts/exchange/thirdparty/PersonalMarket.sol:22:
legacy/contracts/exchange/thirdparty/PersonalMarket.sol:23:    event LogMake(
legacy/contracts/exchange/thirdparty/PersonalMarket.sol:33:
legacy/contracts/exchange/thirdparty/PersonalMarket.sol:34:    event LogBump(
legacy/contracts/exchange/thirdparty/PersonalMarket.sol:44:
legacy/contracts/exchange/thirdparty/PersonalMarket.sol:45:    event LogTake(
legacy/contracts/exchange/thirdparty/PersonalMarket.sol:56:
legacy/contracts/exchange/thirdparty/PersonalMarket.sol:57:    event LogKill(
legacy/contracts/exchange/thirdparty/SimpleMarket.sol:18:    event LogItemUpdate(uint id);
legacy/contracts/exchange/thirdparty/SimpleMarket.sol:19:    event LogTrade(uint pay_amt, address indexed pay_gem,
legacy/contracts/exchange/thirdparty/SimpleMarket.sol:21:
legacy/contracts/exchange/thirdparty/SimpleMarket.sol:22:    event LogMake(
legacy/contracts/exchange/thirdparty/SimpleMarket.sol:32:
legacy/contracts/exchange/thirdparty/SimpleMarket.sol:33:    event LogBump(
legacy/contracts/exchange/thirdparty/SimpleMarket.sol:43:
legacy/contracts/exchange/thirdparty/SimpleMarket.sol:44:    event LogTake(
legacy/contracts/exchange/thirdparty/SimpleMarket.sol:55:
legacy/contracts/exchange/thirdparty/SimpleMarket.sol:56:    event LogKill(
legacy/contracts/exchange/thirdparty/dexy/DexyExchange.sol:6:
legacy/contracts/exchange/thirdparty/dexy/DexyExchange.sol:7:    event Cancelled(bytes32 indexed hash);
legacy/contracts/exchange/thirdparty/dexy/DexyExchange.sol:8:
legacy/contracts/exchange/thirdparty/dexy/DexyExchange.sol:9:    event Traded(
legacy/contracts/exchange/thirdparty/dexy/DexyExchange.sol:18:
legacy/contracts/exchange/thirdparty/dexy/DexyExchange.sol:19:    event Ordered(
legacy/contracts/exchange/thirdparty/dexy/DexyVault.sol:97:
legacy/contracts/exchange/thirdparty/dexy/DexyVault.sol:98:    event Deposited(address indexed user, address token, uint amount);
legacy/contracts/exchange/thirdparty/dexy/DexyVault.sol:99:    event Withdrawn(address indexed user, address token, uint amount);
legacy/contracts/exchange/thirdparty/dexy/DexyVault.sol:100:
legacy/contracts/exchange/thirdparty/dexy/DexyVault.sol:101:    event Approved(address indexed user, address indexed spender);
legacy/contracts/exchange/thirdparty/dexy/DexyVault.sol:102:    event Unapproved(address indexed user, address indexed spender);
legacy/contracts/exchange/thirdparty/dexy/DexyVault.sol:103:
legacy/contracts/exchange/thirdparty/dexy/DexyVault.sol:104:    event AddedSpender(address indexed spender);
legacy/contracts/exchange/thirdparty/dexy/DexyVault.sol:105:    event RemovedSpender(address indexed spender);
legacy/contracts/exchange/thirdparty/SimpleMarketWithApprove.sol:18:    event LogItemUpdate(uint id);
legacy/contracts/exchange/thirdparty/SimpleMarketWithApprove.sol:19:    event LogTrade(uint pay_amt, address indexed pay_gem,
legacy/contracts/exchange/thirdparty/SimpleMarketWithApprove.sol:21:
legacy/contracts/exchange/thirdparty/SimpleMarketWithApprove.sol:22:    event LogMake(
legacy/contracts/exchange/thirdparty/SimpleMarketWithApprove.sol:32:
legacy/contracts/exchange/thirdparty/SimpleMarketWithApprove.sol:33:    event LogBump(
legacy/contracts/exchange/thirdparty/SimpleMarketWithApprove.sol:43:
legacy/contracts/exchange/thirdparty/SimpleMarketWithApprove.sol:44:    event LogTake(
legacy/contracts/exchange/thirdparty/SimpleMarketWithApprove.sol:55:
legacy/contracts/exchange/thirdparty/SimpleMarketWithApprove.sol:56:    event LogKill(
legacy/contracts/exchange/thirdparty/MatchingMarket.sol:7:    event Approval(address indexed src, address indexed guy, uint wad);
legacy/contracts/exchange/thirdparty/MatchingMarket.sol:8:    event Transfer(address indexed src, address indexed dst, uint wad);
legacy/contracts/exchange/thirdparty/MatchingMarket.sol:24:    event LogItemUpdate(uint id);
legacy/contracts/exchange/thirdparty/MatchingMarket.sol:25:    event LogTrade(uint pay_amt, address indexed pay_gem,
legacy/contracts/exchange/thirdparty/MatchingMarket.sol:27:
legacy/contracts/exchange/thirdparty/MatchingMarket.sol:28:                   event LogMake(
legacy/contracts/exchange/thirdparty/MatchingMarket.sol:38:
legacy/contracts/exchange/thirdparty/MatchingMarket.sol:39:                   event LogBump(
legacy/contracts/exchange/thirdparty/MatchingMarket.sol:49:
legacy/contracts/exchange/thirdparty/MatchingMarket.sol:50:                   event LogTake(
legacy/contracts/exchange/thirdparty/MatchingMarket.sol:61:
legacy/contracts/exchange/thirdparty/MatchingMarket.sol:62:                   event LogKill(
legacy/contracts/exchange/thirdparty/MatchingMarket.sol:340:    event LogBuyEnabled(bool isEnabled);
legacy/contracts/exchange/thirdparty/MatchingMarket.sol:341:    event LogMinSell(address pay_gem, uint min_amount);
legacy/contracts/exchange/thirdparty/MatchingMarket.sol:342:    event LogMatchingEnabled(bool isEnabled);
legacy/contracts/exchange/thirdparty/MatchingMarket.sol:343:    event LogUnsortedOffer(uint id);
legacy/contracts/exchange/thirdparty/MatchingMarket.sol:344:    event LogSortedOffer(uint id);
legacy/contracts/exchange/thirdparty/MatchingMarket.sol:345:    event LogAddTokenPairWhitelist(ERC20 baseToken, ERC20 quoteToken);
legacy/contracts/exchange/thirdparty/MatchingMarket.sol:346:    event LogRemTokenPairWhitelist(ERC20 baseToken, ERC20 quoteToken);
legacy/contracts/exchange/thirdparty/MatchingMarket.sol:347:    event LogInsert(address keeper, uint id);
legacy/contracts/exchange/thirdparty/MatchingMarket.sol:348:    event LogDelete(address keeper, uint id);
legacy/contracts/FundInterface.sol:9:
legacy/contracts/FundInterface.sol:10:    event PortfolioContent(address[] assets, uint[] holdings, uint[] prices);
legacy/contracts/FundInterface.sol:11:    event RequestUpdated(uint id);
legacy/contracts/FundInterface.sol:12:    event Redeemed(address indexed ofParticipant, uint atTimestamp, uint shareQuantity);
legacy/contracts/FundInterface.sol:13:    event FeesConverted(uint atTimestamp, uint shareQuantityConverted, uint unclaimed);
legacy/contracts/FundInterface.sol:14:    event CalculationUpdate(uint atTimestamp, uint managementFee, uint performanceFee, uint nav, uint sharePrice, uint totalSupply);
legacy/contracts/FundInterface.sol:15:    event ErrorMessage(string errorMessage);
legacy/contracts/modules/SimpleCertifier.sol:14:
legacy/contracts/modules/SimpleCertifier.sol:15:    event NewOwner(address indexed old, address indexed current);
legacy/contracts/modules/SimpleCertifier.sol:24:    event Confirmed(address indexed who);
legacy/contracts/modules/SimpleCertifier.sol:25:    event Revoked(address indexed who);
legacy/contracts/exchange/thirdparty/0x/Exchange.sol:1709:    event Fill(
legacy/contracts/exchange/thirdparty/0x/Exchange.sol:1724:    event Cancel(
legacy/contracts/exchange/thirdparty/0x/Exchange.sol:1734:    event CancelUpTo(
legacy/contracts/exchange/thirdparty/0x/Exchange.sol:1841:    event AssetProxyRegistered(
legacy/contracts/exchange/thirdparty/0x/Exchange.sol:1897:    event SignatureValidatorApproval(
legacy/contracts/version/VersionInterface.sol:9:
legacy/contracts/version/VersionInterface.sol:10:    event FundUpdated(uint id);
legacy/contracts/version/Version.sol:31:
legacy/contracts/version/Version.sol:32:    event FundUpdated(address ofFund);
legacy/contracts/Fund.sol:20:
legacy/contracts/Fund.sol:21:    event OrderUpdated(address exchange, bytes32 orderId, UpdateType updateType);
legacy/contracts/pricefeeds/PriceFeedInterface.sol:10:
legacy/contracts/pricefeeds/PriceFeedInterface.sol:11:    event PriceUpdated(bytes32 hash);
legacy/contracts/pricefeeds/CanonicalPriceFeed.sol:16:    event SetupPriceFeed(address ofPriceFeed);
legacy/contracts/pricefeeds/SimplePriceFeedInterface.sol:10:
legacy/contracts/pricefeeds/SimplePriceFeedInterface.sol:11:    event PriceUpdated(bytes32 hash);

```

**Comment by schmidsi** [comment]
You can skip timestamps since they are redundant. By observing an event, we get the blocknumber and therefore timestamps.

**Comment by travs** [comment]
Closed by #609
